### ISSUE TITLE
Redefining operators in the same scope ressults in allocations

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimaCore.jl#346]
See [this comment](https://github.com/CliMA/ClimaCore.jl/blob/main/test/fdspaces_opt_examples.jl#L142).

**Comment by jakebolewski** [comment]
I think the issue is that what you linked to is an alias for an abstract type (unionall vector for any FT), if it was a concrete bitstype the allocation should not happen I'm guessing.

This is why it is probably good to be wary of non-concrete aliases like this

**Comment by charleskawczynski** [comment]
Ah, whoops, well, I think [this test](https://github.com/CliMA/ClimaCore.jl/blob/main/test/fdspaces_opt_examples.jl#L180-L214) doesn't use the union all type

**Comment by jakebolewski** [comment]
I guess i can see why one would potentially need to re-define operators with boundary conditions, but ones like these might be better to define once as a module const and use the same operator throughout the code.

I guess my point is maybe we should prioritize where re-defining the operator is not possible.

**Comment by charleskawczynski** [comment]
I completely agree, I don't think this is a long-term issue for TurbulenceConvection.jl-- we should be able to easily fix anywhere that we hit this. I figure it's just helpful to keep the issue open for now so that people are aware.
